frontend/src/components/DistillationTab/StatementAlertsAccordion.tsx

```diff
@@ -0,0 +1,352 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Typography,
+  Box,
+} from "@mui/material";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import StatementForm from "../Forms/StatementForm";
+import connectivityStatementService from "../../services/StatementService";
+import Stack from "@mui/material/Stack";
+import IconButton from "@mui/material/IconButton";
+import { DeleteOutlined} from "@mui/icons-material";
+import statementService from "../../services/StatementService";
+import Select from "@mui/material/Select";
+import AlertMenuItem from "./AlertMenuItem";
+import {vars} from "../../theme/variables";
+import ConfirmationDialog from "./ConfiramtionDialog";
+
+const parseTextWithLinks = (text: string, vars: any): JSX.Element[] => {
+  const urlRegex = /(https?:\/\/\S+|www\.\S+|[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})/gi;
+  
+  return text.split(urlRegex).map((part, index) => {
+    const isURL = urlRegex.test(part);
+    if (isURL) {
+      const href = part.startsWith("http") ? part : `http://${part}`;
+      return (
+        <React.Fragment key={index}>
+          <a
+            key={index}
+            href={href}
+            target="_blank"
+            rel="noopener noreferrer"
+            style={{ color: vars.primary800, textDecoration: "underline" }}
+          >
+            {part}
+          </a>
+        </React.Fragment>
+      );
+    }
+    return <React.Fragment key={index}>{part}</React.Fragment>;
+  });
+};
+const StatementAlertsAccordion = (props: any) => {
+  const { statement, refreshStatement, isDisabled, setStatement } = props;
+  
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const [activeTypes, setActiveTypes] = useState<number[]>([]);
+  const [alerts, setAlerts] = useState<any[]>([]);
+  const [openFormIndex, setOpenFormIndex] = useState<number | null>(null);
+  const [openDialog, setOpenDialog] = useState<boolean>(false);
+  const [alertToDelete, setAlertToDelete] = useState<number | null>(null);
+  
+  const currentAlertRef = useRef<any>(null);
+  const handleChange = (event: React.SyntheticEvent, isExpanded: boolean) => {
+    setExpanded(isExpanded);
+  };
+  
+  const addAlert = (typeId: number) => {
+    if (!activeTypes.includes(typeId)) {
+      const newAlert = { connectivity_statement: parseInt(statement.id), alert_type: typeId, text: "" }
+      
+      connectivityStatementService.createAlert(newAlert).then((res: any) => {
+        currentAlertRef.current = res;
+        
+        const updatedAlerts = [
+          ...(statement.statement_alerts || []),
+          res,
+        ];
+        const updatedStatement = { ...statement, statement_alerts: updatedAlerts };
+        setActiveTypes([...activeTypes, typeId]);
+        setStatement(updatedStatement);
+        const newIndex = updatedAlerts.length - 1;
+        setOpenFormIndex(newIndex);
+        setTimeout(() => {
+          const textArea = document.querySelectorAll(`#root_statement_alerts_0_text`);
```

Comment:
suggestion: I would prefer that we don't rely on a hardcoded id to focus, instead maybe we can have a reference to the statement alerts element and use it here